### PR TITLE
refactor(ui):  add request loading indicator to chat playground

### PR DESCRIPTION
**Pull request by liangfung**
Close TAB-578
1. scroll to bottom after request
2. add request loading indicator
https://jam.dev/c/ad3f8161-25b8-4381-8ad9-2b7dcc05c773

### CODE REVIEW

**Comment by liangfung** [comment]
cc @wwayne

File: ee/tabby-ui/components/chat-list.tsx [hunk 1]
```diff
@@ -27,6 +32,12 @@ export function ChatList({ messages, handleMessageAction }: ChatList) {
           )}
         </div>
       ))}
+      {isStreamResponsePending && (
+        <>
+          <Separator className="my-4 md:my-8" />
+          <MessagePendingIndicator />
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat-list.tsx:38]: should this use a skeleton placeholder for consistency?

File: ee/tabby-ui/components/chat.tsx [hunk 2]
```diff
@@ -122,12 +152,17 @@ function ChatRenderer(
...
-    const scrollHeight = document.documentElement.scrollHeight
-    window.scrollTo(0, scrollHeight)
+    scrollToBottom()
 
     return () => stop()
   }, [])
 
+  React.useLayoutEffect(() => {
+    if (isStreamResponsePending) {
+      scrollToBottom('smooth')
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat.tsx:162]: call to `scrollToBottom` should be debounced (e.g every 5 seconds) to make the UX more friendly

Comment (liangfung) [reply ee/tabby-ui/components/chat.tsx:162]: This callback will only be executed once after appending a request message

Comment (wsxiaoys) [reply ee/tabby-ui/components/chat.tsx:162]: but appending message could happens at higher frequency (e.g much faster than every 5s)

File: ee/tabby-ui/components/chat-list.tsx (hunk 1 above)
Comment (liangfung) [reply ee/tabby-ui/components/chat-list.tsx:38]: image

**Pull request merged**